hashring/checksum.go

```diff
@@ -0,0 +1,67 @@
+package hashring
+
+import (
+	"bytes"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/dgryski/go-farm"
+)
+
+// Checksum computes a checksum for an instance of a HashRing. The
+// checksum can be used to compare two rings for equality.
+type Checksum interface {
```

Comment:
I don't agree here. A function can also implement an interface when needed, but if you want to use a struct it is harder if you don't use an interface as input.

The naming you are right on, we should call the interface `Checksummer`.